frame/transaction-payment/rpc/runtime-api/src/lib.rs

```diff
@@ -30,5 +32,37 @@ sp_api::decl_runtime_apis! {
 	{
 		fn query_info(uxt: Block::Extrinsic, len: u32) -> RuntimeDispatchInfo<Balance>;
 		fn query_fee_details(uxt: Block::Extrinsic, len: u32) -> FeeDetails<Balance>;
+		fn query_call_info(encoded_call: Bytes) -> Result<RuntimeDispatchInfo<Balance>, DispatchError>;
+		fn query_weight_to_fee(encoded_call: Bytes) -> Result<Balance, DispatchError>;
```

Comment:
As option is to be more consistent and instead expose 
```
query_call_fee_details(...) -> Result<RuntimeDispatchInfo<Balance>, DispatchError>
```
Like `query_fee_details` for `Extrinsic`. 
But for our use case, what we really need, is a `weight_to_fee` result. 
No `base_fee`, `len_fee`, `adjusted_weight_fee` (can be also used for our use case).

Reply:
Please implement the general case with consistent naming as you suggested (returning `FeeDetails`). Also using `Bytes` here makes no sense. Use the proper type (`Call`).